pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go

```diff
@@ -160,6 +160,68 @@ type KeycloakAPIRealm struct {
 	// Default Locale
 	// +optional
 	DefaultLocale string `json:"defaultLocale,omitempty"`
+
+	// Roles
+	// +optional
+	Roles *RolesRepresentation `json:"roles,omitempty"`
+}
+
+type RoleRepresentationArray []RoleRepresentation
+
+// https://www.keycloak.org/docs-api/11.0/rest-api/index.html#_rolesrepresentation
+type RolesRepresentation struct {
+	// Client Roles
+	// +optional
+	Client map[string]RoleRepresentationArray `json:"client,omitempty"`
```

Comment:
Could we change this to use arrays instead of type redefinition? The Java type is `Map<String, List<RoleRepresentation>>`, so maybe `map[string][]RoleRepresentation` would somehow work? I'm not sure if this makes the Go compiler happy but you probably know what I'm trying to achieve here.

Reply:
I'd also like that better than what I had to do here.
However `make code/gen` complains
```
.../keycloak-operator/pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go:175:20: map values must be a named type, not *ast.ArrayType
FATA[0000] error generating CRDs from APIs in pkg/apis: error generating CRD manifests: error running CRD generator: not all generators ran successfully 
```
That's the whole reason, also for the discussions above.
The cause is the controller-tools people not wanting this as linked further up https://github.com/kubernetes-sigs/controller-tools/issues/213#issuecomment-507844871
Since we want to replicate the existing data structures as closely as possible this is as close as we can get AFAIK.

Reply:
Ok, I understand.